src/components/mixins/UtilityMixin.vue

```diff
@@ -40,24 +40,17 @@ export default class UtilityMixin extends Vue {
             }
 
             const game = this.$store.state.activeGame;
-            const combo: ThunderstoreCombo | R2Error = await ThunderstoreCombo.fromProtocol(protocolUrl, game);
-            if (combo instanceof R2Error) {
-                this.$store.commit('error/handleError', {
-                    error: combo,
-                    severity: LogSeverity.ACTION_STOPPED
-                });
-                return;
-            }
-
             try {
+                const combo: ThunderstoreCombo = throwForR2Error(await ThunderstoreCombo.fromProtocol(protocolUrl, game));
                 await this.$store.dispatch('download/downloadAndInstallSpecific', {profile: this.profile.asImmutableProfile(), combo});
-                const modList = await ProfileModList.getModList(this.profile.asImmutableProfile());
-                if (modList instanceof R2Error) {
-                    throw modList;
-                }
+                
+                const modList = throwForR2Error(await ProfileModList.getModList(this.profile.asImmutableProfile()));
```

Comment:
This will probably conflict with the changes in #1717 . Up to you whether you want to rebase this stack or handle the merge conflicts later.

Reply:
I rebased the stack 👍 
My original thinking was that, as the changes don't really depend on the #1717 changes, to keep them separated (to avoid blocking).
Now that #1717 is approved, it won't become a blocker, so it's safe to rebase. 👌